### PR TITLE
added polygon data-type

### PR DESCRIPTION
**Pull request by ericchapman**
I didn't write any tests yet and I think the "reverse" code can be slightly simplified from your Point fix.  I'll let you handle that when you add tests.

I did test it by using it.

### CODE REVIEW

File: Sources/PostgreSQL/Data/PostgreSQLData+Polygon.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+/// A 2-dimensional list of (double[2]) points representing a polygon.
+public struct PostgreSQLPolygon: Codable, Equatable {
+    /// The points that make up the polygon.
+    public var points: [PostgreSQLPoint]
+
+    /// Create a new `Point`
```
Comment (tanner0101) [inline Sources/PostgreSQL/Data/PostgreSQLData+Polygon.swift]: typo

File: Sources/PostgreSQL/Data/PostgreSQLData+Polygon.swift [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+extension PostgreSQLPolygon: CustomStringConvertible {
+    /// See `CustomStringConvertible`.
+    public var description: String {
+        return "(\(self.points.map{ $0.description }.joined(separator: ",")))"
+    }
+}
+
+extension PostgreSQLPolygon: PostgreSQLDataConvertible {
+
```
Comment (tanner0101) [inline Sources/PostgreSQL/Data/PostgreSQLData+Polygon.swift]: space

File: Sources/PostgreSQL/Data/PostgreSQLData+Polygon.swift (hunk 1 above)
Comment (ericchapman) [reply Sources/PostgreSQL/Data/PostgreSQLData+Polygon.swift]: fixed

File: Sources/PostgreSQL/Data/PostgreSQLData+Polygon.swift (hunk 2 above)
Comment (ericchapman) [reply Sources/PostgreSQL/Data/PostgreSQLData+Polygon.swift]: fixed

**Comment by ericchapman** [comment]
@tanner0101 Sorry for the 4 commits.  I got in a weird GIT branch issue, had to merge, and then saw your feedback.

**Comment by penny-coin** [comment]
Hey @ericchapman, you just merged a pull request, have a coin! 

You now have 1 coins.

**Pull request merged**